src/classes/RD_RecurringDonations.cls

```diff
@@ -1354,41 +1375,52 @@ public class RD_RecurringDonations {
     * @param rdcounter Opportunity counter for naming and Recurring_Donation_Installment_Number__c
     * @return Opportunity
     */
-    private static Opportunity newOpportunityOnDate(Date oppCloseDate, npe03__Recurring_Donation__c r, Integer rdcounter) {
+    private static Opportunity newOpportunityOnDate(npe03__Recurring_Donation__c r, Date oppCloseDate, Decimal amount, Contact con, Integer rdcounter) {
 
+        Sobjectfield ConIDForRole = Schema.sObjectType.Opportunity.fields.getMap().get('npe01__Contact_Id_for_Role__c');
         npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
 
-        Opportunity o = new Opportunity();
-        o.CloseDate = oppCloseDate;
-        o.OwnerId = r.OwnerId;
+        Opportunity opp = new Opportunity();
+
+        opp.CloseDate = oppCloseDate;
+        opp.OwnerId = r.OwnerId;
+        opp.Amount = amount;
+        opp.Primary_Contact__c = r.npe03__Contact__c;
+        opp.Recurring_Donation_Installment_Number__c = rdcounter;
+        opp.StageName = System.label.npe03.RecurringDonationStageName;
+        opp.npe03__Recurring_Donation__c = r.Id;
 
         if(isMultiCurrency){
             //set the donation currency field to equal the recurring donation currency
-            o.put(OppCurrencyField,r.get(RDCurrencyField));
+            opp.put(OppCurrencyField,r.get(RDCurrencyField));
         }
 
-        o.Amount = r.npe03__Amount__c;
-
-        o.Recurring_Donation_Installment_Number__c = rdcounter;
-
-        o.Name = getOpportunityName(o, r);
+        if ( r.npe03__Organization__c != null ) {
+            opp.AccountId = r.npe03__Organization__c;
+        } else if ( con != null ) {
+            opp.AccountId = con.AccountId;
+            if (ConIDForRole != null){
```

Comment:
Is there a reason that the `npe01__Contact_Id_For_Role__c` would not exist on the Opportunity object?

Reply:
@bethbrains Any thoughts on this one?

Reply:
Ah, my email reply didn't work, sorry.
This is how it was in the legacy code, so I kept it. Not sure if there was a good reason.

Reply:
Hard to tell what's new and what's old code for some of this. There's no reason for it to be this way, so if you're going to do another commit to fix the unit teste I'd change this and delete the code that creates the `ConIDForRole` var.